### PR TITLE
[LimitOrder] Price suggestions refactor

### PR DESCRIPTION
**Pull request by anxolin**
Part of #1381

image

This is the second of several PRs for refactoring the Trading Widget.

## Includes
This PR:
* Isolate into it's own common component:
    * `EllipsisText`
    * `/SwapPrice`
    * `OnchainOrderbookPriceEstimation`
    * `Price`
    * `PriceEstimations`
* It also creates storybooks for all of them.
* Specially for the Price Estimation component, creates some basic stories
* Creates a generic "Frame" component. The component is not meant to be used in the app, it's just a box with some dashed lines to be used as a helper to build stories. You can set the width and has nice defaults.

## Regarding some changes of naming functions in props
I'm changing the imperative naming of some events into a more "event" based name. 

For example when you click on the suggested prices, we used to invoke a function called `updatePrices`, however using an imperative can make our name to be deciding on it's behaviour, and that is against it being general purpouse. 

I've called it `onClickPrice`, then the smart component does:
```tsx
 <PriceSuggestions
      ... 
      onClickPrice={updatePrices}
    />
```

## Note on PriceSuggestionComp.tsx 

* First the name: I've used `Comp` naming convention for the smart components, but happy to agree on other. I chosed `Comp` cause adding `Component` on top of other long component names can be too much. Similar as how we use `Impl` for the sufix of the API `Implementations`
* Second, it's content. I will probably delete this component in a future PR. I'm not 100% sure yet. For now, it makes it easier to let the app keep working as it used to be (same props, and dependency on a smart component), but now we have a dumb version of it. However, in a future PR I will probably move it up in the hierarchy to the smart component for the `LimitOrder`

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr1373--dexreact.review.gnosisdev.com

**Review by W3stside — APPROVED** [review]
We used to just add smart components rendering dumb ones in `containers/ComponentName` no need for comp or whatever added. Or just name the smart one `index` inside the component name folder and the dumb one as the proper name e.g `ComponentName.tsx` <- dumb, `index.tsx` <- smart

**Comment by anxolin** [comment]
> We used to just add smart components rendering dumb ones in containers/ComponentName no need for comp or whatever added. Or just name the smart one index inside the component name folder and the dumb one as the proper name e.g ComponentName.tsx <- dumb, index.tsx <- smart

Both options work for me, I think this Component is meant to disappear in future PRs as I mentioned in the description. I mean to move the smart logic up in the hierarchy if possible

File: src/components/common/EllipsisText.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
+import { Story, Meta } from '@storybook/react/types-6-0'
+
+import { EllipsisText, Props } from 'components/common/EllipsisText'
+
+export default {
+  title: 'Common/EllipsisText',
```
Comment (Velenir) [inline src/components/common/EllipsisText.stories.tsx:8]: Book sad again

File: src/components/common/SwapPrice.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+
+// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
+import { Meta, Story } from '@storybook/react/types-6-0'
+import { SwapPrice, Props } from './SwapPrice'
+
+export default {
+  title: 'Common/SwapPrice',
```
Comment (Velenir) [inline src/components/common/SwapPrice.stories.tsx:8]: Book the saddest it's been

File: src/components/trade/Price.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+// import { Price } from 'components/trade/PriceP'
+import { useForm, FormProvider } from 'react-hook-form'
+
+// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
+import { Meta, Story } from '@storybook/react/types-6-0'
+import { Price, Props } from './Price'
+
+export default {
+  title: 'Trade/Price',
```
Comment (Velenir) [inline src/components/trade/Price.stories.tsx:10]: :book: :cry:

File: src/components/trade/PriceSuggestions/PriceSuggestions.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
+import { Meta, Story } from '@storybook/react/types-6-0'
+import { PriceSuggestions, Props } from './PriceSuggestions'
+import { Frame } from 'components/common/Frame'
+import BigNumber from 'bignumber.js'
+
+export default {
+  title: 'Trade/PriceSuggestions',
+  components: PriceSuggestions,
```
Comment (Velenir) [inline src/components/trade/PriceSuggestions/PriceSuggestions.stories.tsx:12]: Book confused
There is `component` and there is `subcomponents`, no `components`

File: src/services/factories/getTokenFromExchange.ts [hunk 5]
```diff
@@ -21,7 +21,7 @@ interface GetByAddressParams {
   tokenAddress: string
 }
 
-export const enum TokenFromExchange {
+export enum TokenFromExchange {
```
Comment (Velenir) [inline src/services/factories/getTokenFromExchange.ts:24]: I also stepped on this
`babel-loader-typescript` can't handle const enums

**Review by Velenir — APPROVED** [review]

File: src/components/common/SwapPrice.stories.tsx (hunk 2 above)
Comment (anxolin) [reply src/components/common/SwapPrice.stories.tsx:8]: This one was addressed in https://github.com/gnosis/dex-react/pull/1388

File: src/components/trade/Price.stories.tsx (hunk 3 above)
Comment (anxolin) [reply src/components/trade/Price.stories.tsx:10]: I'll address all book sadness in  #1388

File: src/components/trade/PriceSuggestions/PriceSuggestions.stories.tsx (hunk 4 above)
Comment (anxolin) [reply src/components/trade/PriceSuggestions/PriceSuggestions.stories.tsx:12]: I'll address with the other ones

**Comment by Velenir** [comment]
Some stories are broken :books: :fire:

**Pull request merged**